src/m365/search/commands/externalconnection/externalconnection-get.spec.ts

```diff
@@ -0,0 +1,258 @@
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    (command as any).items = [];
+    
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.EXTERNALCONNECTION_GET), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('gets an external connection', (done: any) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/external/connections/TestConnectionForCLI`) {
+        return Promise.resolve(externalConnectionGetResponse);
+      }
+      return Promise.reject('Invalid request');
+    });
+    const options: any = {
+      debug: false,
+      id: 'TestConnectionForCLI',
+      name: 'Test Connection for CLI',
+      description: 'Test connection that will not do anything'
+    };
+
+    command.action(logger, { options: options }, () => {
+      try {
+        assert(loggerLogSpy.calledWith(externalConnectionGetResponse));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('gets an external connection with authorized app id', (done: any) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/external/connections/TestConnectionForCLIWithAppIDs`) {
+        return Promise.resolve(externalConnectionGetResponseWithAppIDs);
+      }
+      return Promise.reject('Invalid request');
+    });
+    const options: any = {
+      debug: false,
+      id: 'TestConnectionForCLIWithAppIDs',
+      name: 'Test Connection for CLI With App IDs',
+      description: 'Test connection that will not do anything'
+    };
+
+    command.action(logger, { options: options }, () => {
+      try {
+        assert(loggerLogSpy.calledWith(externalConnectionGetResponseWithAppIDs));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('fails to get external connection when team does not exists', (done) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      const matchingItemNumber = (opts.url as string).indexOf(`/v1.0/external/connections?$filter=name eq 'sjalfj'`);
+      
+      if (matchingItemNumber > -1) {
+        return Promise.resolve({ value: [] });
+      }
+      return Promise.reject('The specified external connection does not exist');
```

Comment:
This logic belongs in the command not in the test. Right now, the command would always pass without any error, while here we're giving an impression that it will fail.